File: 1497. Check If Array Pairs Are Divisible by k/solution.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {boolean}
 */
var canArrange = function(arr, k) {

    const count = new Map();

    for (let x of arr) {
        let remainder = ((x % k) + k) % k;
        count.set(remainder, (count.get(remainder) || 0) + 1);
    }

    for (let [key, freq] of count.entries()) {
        if (key === 0) {
            if (freq % 2 !== 0) {
                return false;
            }
        }else {
            let complementFreq = count.get(k - key) || 0;
            if (freq !== complementFreq) {
                return false;
            }
        }
    }

    return true;
    
};

File: 1497. Check If Array Pairs Are Divisible by k/solution.cpp
class Solution {
public:
    bool canArrange(vector<int>& arr, int k) {

        unordered_map<int,int>count;

        for(auto &x : arr){
            int remainder = ((x%k)+k)%k; // for handling negative numbers
            count[remainder]++;
        }
            

        for(auto &x : count){
           if(x.first==0){
                if(x.second&1)
                    return false;
           }else if(x.second!=count[k-x.first]){
               return false;
           }
        }
        return true;

    }
};